### PR TITLE
docs: include authentication example to examples page

### LINKED ISSUE #241: Include authentication UI example at examples page
### Which scope/s are relevant/related to the feature request?

Docs

### Information

The current examples page at https://www.spartan.ng/examples/ has only two examples of using spartan/ui components for building UIs.

The goal with this feature/enhancement its adding an authentication example, similar to what we have at shadcn/ui example page at https://ui.shadcn.com/examples/authentication.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by danilolmc**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

The current examples page has very few examples showing what can be done using the spartan/ui library

Closes #241 

## What is the new behavior?

The examples page now gets an authentication example showcasing
what can be done with spartan/ui, this is a starting point to add more examples to the page similar to shadcn/ui

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/app/src/app/pages/(examples)/examples/authentication/(authentication).page.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+@Component({
+	selector: 'spartan-authentication',
+	standalone: true,
+	imports: [AuthenticationFormComponent, HlmButtonDirective, HlmCardDirective],
+	host: {
+		class: 'block',
+	},
+	template: `
+		<div hmlcard class="bg-background overflow-hidden shadow-md md:shadow-xl">
+			<div class="md:hidden">
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(examples)/examples/authentication/(authentication).page.ts:22]: Let's create a responsive version of this screen and remove the shadcn images

File: apps/app/src/app/pages/(examples)/examples/authentication/(authentication).page.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+					src="/assets/auth_example_dark_mobile.png"
+				/>
+			</div>
+			<div
+				class="container relative hidden h-[800px] flex-col items-center justify-center md:grid lg:max-w-none lg:grid-cols-2 lg:px-0"
+			>
+				<a
+					hlmBtn
+					variant="link"
+					class="focus-visible:ring-ring hover:bg-accent absolute right-4 top-4 inline-flex h-9 items-center justify-center whitespace-nowrap rounded-md px-4 py-2 text-sm font-medium  transition-colors hover:no-underline focus-visible:outline-none focus-visible:ring-1 disabled:pointer-events-none disabled:opacity-50 md:right-8 md:top-8"
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(examples)/examples/authentication/(authentication).page.ts:50]: You shouldn't need that many extra classes when using the hlmBtn directive

File: apps/app/src/app/pages/(examples)/examples/authentication/components/form.component.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			<div class="flex flex-col space-y-2 text-center">
+				<h1 class="text-2xl font-semibold tracking-tight">Create an account</h1>
+				<p class="text-muted-foreground text-sm">Enter your email below to create your account</p>
+			</div>
+			<div class="grid gap-6">
+				<form (ngSubmit)="send()">
+					<div class="grid gap-2">
+						<div class="grid gap-1">
+							<label
+								class="sr-only text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(examples)/examples/authentication/components/form.component.ts:29]: Can we use hlmLabel for this?

File: apps/app/src/app/pages/(examples)/examples/authentication/components/form.component.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+								for="email"
+							>
+								Email
+							</label>
+							<input
+								class="flex h-9 w-full focus-visible:ring-1"
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(examples)/examples/authentication/components/form.component.ts:35]: Do we need these even with the hlmInput?

File: apps/app/src/app/pages/(examples)/examples/authentication/components/form.component.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+								autocomplete="off"
+								type="email"
+							/>
+						</div>
+
+						<button
+							hlmBtn
+							[disabled]="isLoading()"
+							class="h-9 px-4 py-2 focus-visible:ring-1"
+							animate-spin
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(examples)/examples/authentication/components/form.component.ts:50]: animate-spin is normally a class that's added to the icon. I think we can remove it here

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Some minor changes! Thanks for all your work!

**Comment by thatsamsonkid** [comment]
Hi @danilolmc sorry I know this has been sitting for a while, was curious if you could rebase the branch so at least the checks can run again. I think you resolved all the previous comments so I'd be happy to see to getting this merged in

**Comment by danilolmc** [comment]
Hi @thatsamsonkid, no problem, I'm going to rebase the branch.

**Comment by danilolmc** [comment]
It's done @thatsamsonkid !

File: apps/app/src/app/pages/(examples)/examples/authentication/(authentication).page.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import { RouteMeta } from '@analogjs/router';
+import { Component } from '@angular/core';
+import { HlmButtonDirective } from '@spartan-ng/ui-button-helm';
+import { HlmCardDirective } from '@spartan-ng/ui-card-helm';
+import { metaWith } from '~/app/shared/meta/meta.util';
```
Comment (thatsamsonkid) [inline apps/app/src/app/pages/(examples)/examples/authentication/(authentication).page.ts:5]: @danilolmc - this tilde seems to be causing the build to fail, I think this just needs to be updated to "@spartan-ng/app/shared/meta/meta.util" instead.

**Comment by danilolmc** [comment]
Thanks for your review @thatsamsonkid!

Now it's resolved!

**Pull request merged**